File: LeetCode/0342. Power of Four/solution.cpp
/**
 * 1061 / 1061 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 6 MB 
 */
class Solution {
public:
    bool isPowerOfFour(int n) {
        if (n <= 0) return false;
        double res = log(n) / log(4.0);
        return (res - int(res)) < 1e-6;
    }
};

File: LeetCode/0342. Power of Four/solution.py
"""
1061 / 1061 test cases passed.
Runtime: 32 ms
Memory Usage: 14.8 MB
"""
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n <= 0:
            return False
        res = math.log(n) / math.log(4)
        return (res - int(res)) < 1e-6
